ops/framework.py

```diff
@@ -650,13 +651,9 @@ def __init__(
 
         # Flag to indicate that we already presented the welcome message in a debugger breakpoint
         self._breakpoint_welcomed: bool = False
-
-        # Parse the env var once, which may be used multiple times later
-        debug_at = os.environ.get('JUJU_DEBUG_AT')
-        if debug_at:
-            self._juju_debug_at = {x.strip() for x in debug_at.split(',')}
-        else:
-            self._juju_debug_at: Set[str] = set()
+        if juju_debug_at is None:
+            juju_debug_at = _JujuContext.from_dict(os.environ).debug_at
+        self._juju_debug_at = juju_debug_at
```

Comment:
This is the most backwards compatible way of doing this, and if you prefer then leave it like it is.

I think we could simplify it and just have:

```suggestion
        self._juju_debug_at = juju_debug_at or set()
```

Technically, `Framework` is public and we don't tell people not to instantiate it themselves in the docs, but it seems super unlikely that anyone is doing so, or at least doing so and also expecting debug-at values to be loaded.

What about this?

```suggestion
        if juju_debug_at is None:
            warnings.warn("In the future, a `None` value for `juju_debug_at` will have the same meaning as the empty set", warnings.DeprecationWarning)
            juju_debug_at = _JujuContext.from_dict(os.environ).debug_at
        self._juju_debug_at = juju_debug_at
```

Reply:
Yeah, good point. I strongly suspect we could get away with just the `juju_debug_at or set()` version.

Reply:
Got it, I have updated it to `juju_debug_at or set()`, let me run a final test before merging, especially the tests with other charms.